### PR TITLE
feat(models): allow non-integer audit values and ref weights

### PR DESCRIPTION
**Pull request by matejchalk**
Closes:
- code-pushup/portal#152
- code-pushup/portal#391

### CODE REVIEW

**Review by vmasek — DISMISSED** [review]

**Review by BioPhoton — DISMISSED** [review]
LGTM

File: packages/models/src/lib/audit-output.ts
Comment (Tlacenka) [inline packages/models/src/lib/audit-output.ts:1]: 🧪 Coverage: We have [unit tests](https://github.com/code-pushup/cli/blob/main/packages/models/src/lib/audit-output.unit.test.ts) for `audit-output` where the schema is used. Please replace the integer values by floats in some cases to showcase the type of value accepted by the schema.

File: packages/models/src/lib/implementation/schemas.ts [hunk 1]
```diff
@@ -140,7 +142,7 @@ export function packageVersionSchema<TRequired extends boolean>(options?: {
 }
 
 /** Schema for a weight */
-export const weightSchema = nonnegativeIntSchema.describe(
+export const weightSchema = nonnegativeNumberSchema.describe(
```
Comment (Tlacenka) [inline packages/models/src/lib/implementation/schemas.ts:145]: 🧪 Coverage: Similarly, please replace some weights in a `group` or `category-config` schema with floats.

**Review by Tlacenka — APPROVED** [review]
Beautiful 🌈 🧪

**Pull request merged**